lib/sles4sap/sap_deployment_automation_framework/deployment_connector.pm

```diff
@@ -437,18 +437,20 @@ sub destroy_orphaned_peerings {
     );
     my @deleted_peerings;
 
+    my $ret = 0;
     for my $peering (@{$disconnected_peerings}) {
         # Do not delete peering if group it belongs to still exists
         next if az_group_exists(resource_group => $peering->{workload_resource_group}, quiet => 'yes') eq 'true';
-        az_network_peering_delete(
+        $ret = az_network_peering_delete(
             resource_group => $deployer_resource_group,
             vnet => $vnet_name,
             name => $peering->{peering_name});
+        next if ($ret);
         push @deleted_peerings, $peering->{peering_name};
     }
 
     record_info('Peer clean', "Following orphaned peerings were deleted:\n" . join("\n", @deleted_peerings) . "\n");
-    return (\@deleted_peerings);
+    return $ret;
```

Comment:
`$ret` will have the return value from the last call to `az_network_peering_delete`, but if it was called more than once and any of the previous calls failed, then `destroy_orphaned_peerings` will incorrectly return that the peerings were destroyed, even when it wasn't. I think you need 2 variables for the return values: one for each individual call to `az_network_peering_delete` and another for the combination of all the calls.